### ISSUE TITLE
Activation as user service

### ISSUE THREAD

**Issue by gkleen** [nix-community/home-manager#3415]
### Description

I manage a system that has it's root fs on tmpfs and mounts user homes on login via pam_mount over cifs.  
I use the nixos module to manage user environments.

The `home-manager-${userName}.service` units seem to run on boot, far before any user has logged in and their home directory becomes available.

It would be helpful if there was a builtin way to have home-manager activation occur as a systemd *user* service as early as possible, probably before/as a part of `basic.target`?  
Maybe that should even be the default way to do activation for the nixos module?

**Comment by Aleksanaa** [comment]
Also need this. I'm using pam_mount with luks encrypted user home partition and home-manager-${user}.service started too early.

**Comment by ncfavier** [comment]
There's a PR at https://github.com/nix-community/home-manager/pull/2548, @Aleksanaa could you try that?

Out of curiosity, why not mount your LUKS partition at boot? Does using pam_mount mean that you don't have to type your password twice?

**Comment by ncfavier** [comment]
That makes sense. Yeah, I think we're still very much figuring these things out in NixOS, e.g. secure boot is [very recent work in progress](https://x86.lol/generic/2022/11/26/lanzaboote.html).

**Comment by Aleksanaa** [comment]
> There's a PR at #2548, @Aleksanaa could you try that?
> 
> Out of curiosity, why not mount your LUKS partition at boot? Does using pam_mount mean that you don't have to type your password twice?

Anyway, is there an easy way of applying a pr? I'm using flake and home-manager as nixos module.
Do I have to specify inputs to that branch?

**Comment by ncfavier** [comment]
Yeah, add `inputs.home-manager.url = "github:pasqui23/home-manager/nixos-late-start";` to your flake.nix

**Comment by Aleksanaa** [comment]
> Yeah, add `inputs.home-manager.url = "github:pasqui23/home-manager/nixos-late-start";` to your flake.nix

Everything works flawlessly for me!

**Comment by gkleen** [comment]
#2548 seems like the way to go, thanks!

**Comment by Aleksanaa** [comment]
This is still very important for us. Without activation as user service, we can't easily handle more complicated situations of home directory.

**Comment by gkleen** [comment]
My situation wrt. to this issue hasn't changed  
I'm currently having to do extra work whenever I want to upgrade my systems because I can't track home-manager releases and/or master  
Would love to see #2548 merged

**Comment by Aleksanaa** [comment]
?

**Comment by nakoo** [comment]
Please consider to apply this change.

**Comment by mio-19** [comment]
> Also need this. I'm using pam_mount with luks encrypted user home partition and home-manager-${user}.service started too early.

I have the same issue

I am using seperated encrypted user home with zfs native encryption

**Comment by agvantibo-again** [comment]
Basically, every use case that mounts the home directory after authentication relies on this being an option.
